### DISCUSSION TITLE
ESP32 Minimal Bluetooth Build just scanning

### DISCUSSION THREAD

**Discussion by SL2021-Dev** [micropython/micropython#12168] [ESP32]
I need to be able to use OTA but Bluetooth is bloated.
All I need is to sacn for BLE devices no more.
In the past I used...
`def bt_irq(event, data):
	if event == _IRQ_SCAN_RESULT:
	     print(bytes(data[4])) 
`

Changing... lv_micropython\ports\esp32\mpconfigport.h
![image](https://github.com/micropython/micropython/assets/139779102/903e7fc0-cd7a-46d7-8d72-9f9f17d42cd4)
![image](https://github.com/micropython/micropython/assets/139779102/19cfc12f-dbdf-48d5-967c-0260cea7d9c6)

I am using the lv_micropython version that came in @ 2.1MB !!!!
Changing the module selection as shown above I come in @ 1.875MB
This is a 226K Lighter Build !!!
Any other combination, like above or the default, will not build.
Also... could someone explain those combinations.
Can I request **NIMBLE** directly??
Examples, Ideas... anythig would be appreciated !!!

**Comment by gitcnd** [comment]
See here maybe?  https://github.com/orgs/micropython/discussions/12818
I'm using

 ```
"Firmware (Support for OTA)"
[v1.21.0 (2023-10-05) .bin](https://micropython.org/resources/firmware/ESP32_GENERIC-OTA-20231005-v1.21.0.bin) (latest) 
```

from https://micropython.org/download/ESP32_GENERIC/

**Comment by jimmo** [comment]
@SL2021-Dev You must enable `MICROPY_BLUETOOTH_NIMBLE` to use BLE on ESP32. Unfortunately we don't currently have a way to disable peripheral role or gatt server. (You can only disable central and/or gatt client)... the thinking there was that most devices are peripherals & servers, generally the central & client would be a phone/pc.

But I agree, it would be useful to make this more flexible. (Essentially you want just the "observer" role and no GATT support at all).

You can disable pairing & bonding. You must not disable `MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS` or `MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS_WITH_INTERLOCK` on ESP32 or it will not work correctly. Similarly you must use `MICROPY_BLUETOOTH_NIMBLE_BINDINGS_ONLY` because the IDF provides the NimBLE implementation.

I'm not sure what you mean by "Can I request NimBLE directly" sorry.

**Reply by SL2021-Dev** [reply]
So I set the following...
![image](https://github.com/micropython/micropython/assets/139779102/82724bac-4a9b-440f-b746-d5bfc0837c02)
and got...
![image](https://github.com/micropython/micropython/assets/139779102/9091df6e-a094-400b-a59f-49cc07fb927e)

This is really disappointing I can't reduce this in any way.
